aayebare/node-backend-Ekalaamu
# #168398283 Email verification

## What does this PR do?
- Adds email verification functionality for newly registered users

## What are the steps taken
- create email verification route
- modify routes signup, login
- install sendgrid library, mocha, chai, istanbul
- write tests

## How to test it manually
- Register a new user by making a POST request to `/signup` with the required fields and a valid email
- Check the  email you have used to register
<img width="1039" alt="Screen Shot 2019-10-08 at 4 00 32 PM" src="https://user-images.githubusercontent.com/36928255/66397580-cea13f00-e9e4-11e9-9009-44531eb1c3e5.png">

- Click on the link in the email and after it gets redirected from send grid's url, copy and paste it in `POSTMAN`.
- Make a `PUT` request and should a response like this:
![Screen Shot 2019-10-08 at 4 16 16 PM](https://user-images.githubusercontent.com/36928255/66398744-fd201980-e9e6-11e9-84a1-de3272323798.png)


## Relevant PT stories
168398283
